synapse/storage/push_rule.py

```diff
@@ -184,11 +185,23 @@ def _bulk_get_push_rules_for_room(self, room_id, state_group, current_state_ids,
             if uid in local_users_in_room:
                 user_ids.add(uid)
 
+        forgotten = yield self.who_forgot_in_room(
+            event.room_id, on_invalidate=cache_context.invalidate,
+        )
```

Comment:
Why do we need to check if the user has forgotten the room? I guess it's because the user might have left the room and forgotten it but we are still getting new messages with them joined to the room over federation?
Should we be checking whether the user is currently in the room instead?

I'd really like to have some comments here to explain what we are doing and why.